app/src/main/java/com/github/bgabriel998/softwaredevproject/general/SettingsMapActivity.java

```diff
@@ -0,0 +1,186 @@
+        try {
+            json.put("BoundingBox", selectedPoint.computeBoundingBox(GeonamesHandler.DEFAULT_RANGE_IN_KM));
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Handles downloading of the elevation map and the POI list and then adds.
+     * them to the input json. The part regarding the POIs is nested inside this method.
+     *
+     * @param json          json that will contain the bounding box.
+     * @param selectedPoint selected point around which the offline content will be downloaded.
+     */
+    @SuppressLint("StaticFieldLeak")
+    private void addMapAndPOIsToJson(JSONObject json, Point selectedPoint) {
+        new DownloadTopographyTask(){
+            @SuppressLint("StaticFieldLeak")
+            @Override
+            public void onResponseReceived(Pair<int[][], Double> topography) {
+                Log.d("Debug", "entrance");
+                super.onResponseReceived(topography);
+                try {
+                    json.put("ELevationMap", topography);
+                    addPOIsToJson(json, selectedPoint);
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        }.execute(selectedPoint);
+    }
+
+    /**
+     * Handles downloading of the the POI list and then adds
+     * them to the input json. After this process is finished the activity
+     * is terminated.
+     *
+     * @param json          json that will contain the bounding box.
+     * @param selectedPoint selected point around which the offline content will be downloaded.
+     */
+    @SuppressLint("StaticFieldLeak")
+    private void addPOIsToJson(JSONObject json, Point selectedPoint) {
+        new GeonamesHandler(selectedPoint){
+            @Override
+            public void onResponseReceived(ArrayList<POI> result) {
+                List<POIPoint> POIPoints = new ArrayList<>();
+                if(result!=null){
+                    for(POI poi : result){
+                        POIPoint poiPoint = new POIPoint(poi);
+                        poiPoint.setHorizontalBearing(selectedPoint);
+                        poiPoint.setVerticalBearing(selectedPoint);
+                        POIPoints.add(poiPoint);
+                    }
+                }
+
+                try {
+                    json.put("POIPoints", POIPoints);
+                    saveJson(json);
+                    thisActivity.finish();
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+
+            }
+        }.execute();
+    }
+
+    /**
+     * Saves the json file as a .txt.
+     *
+     * @param json  json to save.
+     */
+    private void saveJson(JSONObject json) {
+        String jsonString = json.toString();
+        try {
+            OutputStreamWriter outputStreamWriter = new OutputStreamWriter(openFileOutput("save.txt", Context.MODE_PRIVATE));
```

Comment:
A more meaningful name could be choosen for the `.txt` file. 
Why is the file extension `.txt` and not `.json` ?

Reply:
Because I am saving a `String` and not a `json`. Also, it is very easy to load the `String` back into the `json` as you can do it as `JSONOBbject` `json' = new JSONOBject(stringToLoad)`

Reply:
Ok fine !